islandora_compound_object.module

```diff
@@ -268,17 +268,94 @@ function islandora_compound_object_access(AbstractObject $object) {
 }
 
 /**
- * Replaces parent object thumbnail with child's thumbnail.
+ * Implements hook_compound_cmodel_islandora_derivative().
  */
-function islandora_compound_object_update_parent_thumbnail($parent) {
+function islandora_compound_object_compound_cmodel_islandora_derivative(AbstractObject $object = NULL, $ds_modified_params = array()) {
```

Comment:
I tested and it worked... Odd.

Yeah you are right it should be as you say, so changed!